### ISSUE TITLE
📝 Tailwind to Webstudio Copy/Paste

### ISSUE THREAD

**Issue by itsNintu** [webstudio-is/webstudio#2651]
This feature will enable the direct copying of Tailwind CSS templates and layouts into Webstudio’s interface.

**Comment by brandondrew** [comment]
Will TailwindCSS _**NOT**_ be supported on Webstudio for some reason?  It would be much nicer to have the option to _**use**_ Tailwind rather than only be able to **_convert from_** Tailwind.

**Comment by kof** [comment]
@brandondrew tailwind is literally not necessary in webstudio, it is solving problems with authoring css in code, we don't have the same problems, webstudio design tokens solve most of them already, we just need an easy way to reuse stuff that exists written in tailwind

**Comment by brandondrew** [comment]
Yes, but if (a) I'm already familiar with Tailwind and (b) I might export my site to be hosted elsewhere, then that might not help me.  

One thing I like so much about Webstudio is that you're not locking anyone to your product.  But if you don't support using the tools we already know and love, that's going to be a big detraction.

**Comment by kof** [comment]
> Yes, but if (a) I'm already familiar with Tailwind and (b) I might export my site to be hosted elsewhere, then that might not help me.

Yes I agree generally with the sentiment, though bringing coding habbits to a visual platform that tries to get rid of the problems with code is kinda not a good idea. 

On the other hand exporting to tailwind is def not a bad idea because that's where coding habbits are useful again if someone wants to take the exported code and use it.

Since tailwind classes a really super generic, we can potentially import and export while we don't need to be working with them visually

**Comment by brandondrew** [comment]
> Yes I agree generally with the sentiment, though bringing coding habbits to a visual platform that tries to get rid of the problems with code is kinda not a good idea.

Okay, this is confusing to me.  Perhaps I misunderstood what you're trying to build.   What do you consider "the problems with code"?

**Comment by kof** [comment]
Tailwind is a hack that solves limitations around authoring CSS. You don't have that problem in Webstudio!

**Comment by brandondrew** [comment]
Well, I haven't used Webstudio, so I'll reserve judgement, but that's a pretty extremely negative opinion of something that a lot of people find very useful.  Maybe you're right, but I certainly can't see it yet.

**Comment by kof** [comment]
@brandondrew I don't mean it emotionally negative, I am saying it from technical perspective.

**Comment by gumush** [comment]
+1 Vote

**Comment by Zmove** [comment]
Don't really see the added value of using tailwind in a Webstudio project. Projects that can really take benefits from using tailwind are molecular components based project (generally pretty big ones). Using tailwind outside of that scope is just adding complexity and difficulty to maintain.

I understand kof when he say that Webstudio nature make the use of tailwind useless. Sure it could be a useful addition for some people wanting to use tailwind against all odd, but there are soooo many more important more important to develop that it would be a very very tiny priority to me

**Comment by kof** [comment]
@Zmove primary objective is ability to copy-paste tailwind with html/jsx, which is a handy way to use all the ecosystem of tailwind templates

**Comment by brandondrew** [comment]
Your attitude that normal web technologies are "hacks" and everyone should convert to doing everything your "visual" way is not yielding great success.  I recommend that you embrace the technologies of HTML CSS & JavaScript and stop viewing real code as a problem.  There is no product ever that has succeeded with this attitude as far as I know.  

All No-Code projects create unnecessary barriers for the people that use them. 

The sooner you are willing to interoperate with *normal* technological approaches, the more success WebStudio will have.

**Comment by brandondrew** [comment]
There are all sorts of people in GitHub issues whose basic message is "this feature is not useful TO ME, and therefore it IS NOT USEFUL TO ANYONE".   I hope the world can see how horribly flawed that logic is.  If it's not useful to you, _don't use it_, but please don't try to stop other people from using it.  Such behavior is unproductive.

The more WebStudio can interoperate with normal & widespread web technologies, the more successful it is likely to be.  Conversely, the more it is viewed as a walled garden by those creating it, the more it will be treated as a walled garden that others choose not to enter.  

Treating a hugely successful and widely used set of CSS classes as a "hack" suggests a hostile attitude toward normal code.  Tailwind is really just CSS, designed to be used within HTML, that uses a little JavaScript for tree shaking so you don't need to import all sorts of classes you don't use.

I hope you can make WebStudio interoperate smoothly with code written elsewhere.  That will by far make it the most useful that it can be.

**Comment by brandondrew** [comment]
> Don't really see the added value of using tailwind in a Webstudio project. 

The fact that a feature is not useful to you does not mean it is useful to others.
